### ISSUE TITLE
async await downloads empty content

### ISSUE THREAD

**Issue by alex-drocks** [jmaister/excellentexport#449]
Hello and thanks so much for this nice library.

EDIT: Please see my comment bellow. I found a hacky way to do it.

I cannot understand how to make it work with an **async function as the onClick handler**.
The downloaded file is empty because I am lazy loading the ExcellentExport module with JavaScript Dynamic Import()

### Here's my anchor tag:
```javascript
<a href="#"
   download="list.xlsx"
   id="excelBtn"
   onClick={exportToExcel}
>
  //<button>
  //<svg>
</a>
```

### And here's my exportToExcel async function:
```Javascript
async function exportToExcel() {
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports
  // https://dmitripavlutin.com/javascript-module-import-twice/
  // If a module is imported multiple times, but with the same specifier (i.e. path),
  // the JavaScript specification guarantees that you’ll receive the same module instance.
  console.time("DynamicImport-ExcellentExport");
  const ExcellentExport = await import('excellentexport');
  console.timeEnd("DynamicImport-ExcellentExport");

  return ExcellentExport.convert({
      anchor: "excelBtn",
      format: 'xlsx',
      filename: 'taxes_finance-d_com',
    },
    [
      {
        name: 'taxes.finance-d.com',
        from: {
          table: document.querySelector(".results-table table")
        },
        removeColumns: [7],
        fixValue: function (value, row, column) {
          let fixedValue = value;
          if (column > 4 && value.includes("&nbsp;")) {
            fixedValue = value.replace("&nbsp;", " ");
          }
          return fixedValue;
        }
      }
    ]
  );
}
```

### With this setup, the list.xlsx file is downloaded immediately and the content is empty.
Edit: Actually I don't know if it's empty but Excel says its invalid and might be corrupt so it can't open.

If I remove the download property of the <a> anchor element, it will work but only on the second click on the anchor element.
The first click just navigates to the "#" href in the window.location url.

Can you please give me your insights on this specific use case and if you tihnk of something better than what I'm doing right now?

Thanks in advance,
Alexandre Desroches

**Comment by alex-drocks** [comment]
I found a very unintuitive way to do it.
It could close my issue if you agree that there's nothing else to do around that.

### Using an invisible <a> Anchor element inside <button> (in Next JS)
```Javascript
let ExcellentExport;
//...
        <button
          id="excelBtn"
          title="Exportez la liste de vos calculs dans un fichier compatible avec Excel"
          onClick={async function dynamicImportExcellentExport() {
            // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports
            // https://dmitripavlutin.com/javascript-module-import-twice/
            // If a module is imported multiple times, but with the same specifier (i.e. path),
            // the JavaScript specification guarantees that you’ll receive the same module instance.
            console.time("DynamicImport-ExcellentExport");
            ExcellentExport = await import('excellentexport');
            console.timeEnd("DynamicImport-ExcellentExport");

            const excelBtnAnchor = document.getElementById("excelBtnAnchor");
            excelBtnAnchor.click();  // <-- Trigger the download
          }}
        >
          Excel
          <a href="#"
             id="excelBtnAnchor"
             style={{display: "none"}}
             download="taxes_finance-d_com.xlsx"
             onClick={(e) => {
               e.stopPropagation(); // <-- VERY IMPORTANT (without this infinite download loop)
               return exportToExcel(ExcellentExport);
             }}
          />
        </button>
```

### exportToExcel() function to handle the <a> Anchor click event:
```Javascript
function exportToExcel(ExcellentExport) {
  return ExcellentExport.convert({
      anchor: document.getElementById("excelBtnAnchor"),
      filename: 'taxes_finance-d_com',
      format: 'xlsx',
    },
    [
      {
        name: 'taxes.finance-d.com',
        from: {
          table: document.querySelector(".results-table table")
        },
        removeColumns: [7],
        fixValue: function (value, row, column) {
          let fixedValue = value;
          if (column > 4 && value.includes("&nbsp;")) {
            fixedValue = value.replace("&nbsp;", " ");
          }
          return fixedValue;
        }
      }
    ]
  );
}
```

**Comment by jmaister** [comment]
Fixed with the same like #408.

**Comment by alex-drocks** [comment]
> Fixed with the same like #408.
> 
> Ready on version 3.7.0

Thank you jmaister!!
